### PR TITLE
Feature/cleanup atom sync

### PR DESCRIPTION
**Pull request by talekhinezh**

### CODE REVIEW

File: radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java [hunk 1]
```diff
@@ -749,6 +765,10 @@ public Result transferTokens(
 		return this.execute(transferTokensAction);
 	}
 
+	private long generateTimestamp() {
```
Comment (Sajjon) [inline radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java]: maybe name it `timeNow()` ? :) or `currentTime()`?

File: radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java [hunk 2]
```diff
@@ -749,6 +765,10 @@ public Result transferTokens(
...
+		return System.currentTimeMillis();
+	}
+
 	/**
 	 * Returns an unsigned atom with the appropriate fees given a list of
```
Comment (Sajjon) [inline radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java]: update documentation of method, says it returns `"...an unsigned atom..."`, but we are signing it :)

File: radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java [hunk 3]
```diff
@@ -96,13 +96,29 @@
...
 			this.updates = updates;
-			this.completable = completable;
+			this.completable = updates
+				.ofType(SubmitAtomStatusAction.class)
+				.lastOrError()
+				.flatMapCompletable(status -> {
+					if (status.getStatusNotification().getAtomStatus() == AtomStatus.STORED) {
+						return Completable.complete();
+					} else {
+						JsonElement data = status.getStatusNotification().getData();
```
Comment (Sajjon) [inline radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java]: I know this is not a change, but it was something I reacted to earlier reading the code, it feels a little bit low level to be working with JsonElement at this stage, in RadixApplicationAPI. In this scope the JsonElement is translated into an error, should we maybe try to create an enum holding errors and we working with those instead?

Comment (talekhinezh) [reply radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java]: yeah, good point, I'll add a TODO comment to fix this in the future

File: radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java (hunk 2 above)
Comment (talekhinezh) [reply radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java]: Hmm? the atom returned in the method is still unsigned?

File: radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java [hunk 4]
```diff
@@ -96,13 +96,29 @@
 public class RadixApplicationAPI {
 	private static final Logger LOGGER = LoggerFactory.getLogger(RadixApplicationAPI.class);
 
-	public static class Result {
+	public class Result {
```
[the remark this answers is not in the document]
Comment (talekhinezh) [reply radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java]: Yeah, pretty nasty. But good compromise for now.

**Pull request merged**